datalad/support/dsconfig.py

```diff
@@ -0,0 +1,319 @@
+        for k in self._store:
+            sec, opt = cfg_sectionoption_regex.match(k).groups()
+            if sec == section and opt == option:
+                return True
+        return False
+
+    def get(self, section, option):
+        """Get an option value for the named section"""
+        return self._store['.'.join((section, option))]
+
+    def getint(self, section, option):
+        """A convenience method which coerces the option value to an integer"""
+        return int(self.get(section, option))
+
+    def getfloat(self, section, option):
+        """A convenience method which coerces the option value to a float"""
+        return float(self.get(section, option))
+
+    # this is a hybrid of ConfigParser and dict API
+    def items(self, section=None):
+        """Return a list of (name, value) pairs for each option
+
+        Optionally limited to a given section.
+        """
+        if section is None:
+            return self._store.items()
+        return [(k, v) for k, v in self._store.items() if cfg_section_regex.match(k).group(1) == section]
+
+    #
+    # Compatibility with GitPython's ConfigParser
+    #
+    def get_value(self, section, option, default=None):
+        """Like `get()`, but with an optional default value
+
+        If the default is not None, the given default value will be returned in
+        case the option did not exist. This behavior immitates GitPython's
+        config parser.
+        """
+        try:
+            return self.get(section, option)
+        except KeyError as e:
+            # this strange dance is needed because gitpython does it this way
+            if default is not None:
+                return default
+            else:
+                raise e
+
+    #
+    # Modify configuration (proxy respective git-config call)
+    #
+    def _require_location(self, where, args=None):
+        if args is None:
+            args = []
+        if not where in ('dataset', 'local', 'global'):
+            raise ValueError(
+                "unknown configuration label '{}' (not 'dataset', or 'global')".format(
+                    where))
+        if where == 'dataset':
+            if not self._dataset:
+                raise ValueError(
+                    'ConfigManager cannot store to configuration to dataset, none specified')
+            # create an empty config file if none exists, `git config` will
+            # fail otherwise
+            dscfg_dirname = opj(self._dataset.path, '.datalad')
+            dscfg_fname = opj(dscfg_dirname, 'config')
+            if not exists(dscfg_dirname):
+                os.makedirs(dscfg_dirname)
+            if not exists(dscfg_fname):
+                open(dscfg_fname, 'w').close()
+            args.extend(['--file', opj(self._dataset.path, '.datalad', 'config')])
+        elif where == 'global':
+            args.append('--global')
+        elif where == 'local':
+            args.append('--local')
```

Comment:
aha (just a note to myself) -- so in those cases (local, global) it would deal with git (locally or globally) configuration. good (I guess).  And we would need to not forget to always prefix with "datalad" our variables regardless of the location